source/ShopPanel.h

```diff
@@ -58,9 +81,8 @@ class ShopPanel : public Panel {
 	virtual int DividerOffset() const = 0;
 	virtual int DetailWidth() const = 0;
 	virtual int DrawDetails(const Point &center) = 0;
-	virtual bool CanBuy(bool checkAlreadyOwned = true) const = 0;
-	virtual void Buy(bool alreadyOwned = false) = 0;
-	virtual void FailBuy() const = 0;
+	virtual const BuyResult CanBuy(bool checkAlreadyOwned = true) const = 0;
+	virtual void Buy(bool checkAlreadyOwned = false) = 0;
```

Comment:
This is another point of confusion I guess. So yes after looking at `Buy` closely it probably should be `alreadyOwned`. Maybe `onlyAlreadyOwned`. I was only making the name match `CanBuy`, as the same value is passed to both, but it seems it is used differently in `Buy`.

For example, 'b' will buy an outfit from the store - but will prefer to install from storage/cargo if possible. But if it's not available in the store, CanBuy returns false as it doesn't check cargo - though it would've used cargo to install, if only the store had it available! So the usage of the boolean in CanBuy/Buy is inconsistent. 'true' in CanBuy extends the coverage from checking the store to also check cargo. 'true' in Buy means it only checks cargo. So the name of the boolean is perhaps different for a reason, but still confusing. To fix this confusion I was also considering combining not just FailBuy/CanBuy but also Buy itself (with a boolean argument to actually perform the exchange or just test its availability).

Another example. Hitting 'i' expecting to buy something only in store pops no error message - because CanBuy passed the check (it's in the store!) but Buy didn't buy it (it's not in cargo!)

This is one of many little quirks of the code here. It was clearly designed for simple buy/sell when we now have cargo+storage exchanges. So what exactly 'b' or 'i' means is unclear. It's not just 'buy' as it will install from cargo. 'i' will install, but won't pay to install something.

And at a certain point there could be a giant overhaul, with a more clear source/target of the transaction. That'd be quite a project.